File: project-1-migration/python-sample/src/strings.py
"""String helpers (migration sample)."""


def slugify(text):
    return "-".join(text.strip().lower().split())


def truncate(text, limit=20):
    if len(text) <= limit:
        return text
    return text[:limit].rstrip() + "..."


def count_words(text):
    return len(text.split())

File: project-1-migration/js-sample/src/strings.mjs
// MIGRATED: ESM (after migration)
export function slugify(text) { return text.trim().toLowerCase().split(/\s+/).join("-"); }
export function truncate(text, limit = 20) {
  if (text.length <= limit) return text;
  return text.slice(0, limit).trimEnd() + "...";
}
export function countWords(text) { return text.split(/\s+/).filter(Boolean).length; }
